File: app/src/main/java/life/wanderinglocal/WLApplication.java
package life.wanderinglocal;

import android.app.Application;

import androidx.room.Room;

import java.util.HashSet;

import life.wanderinglocal.repo.FavoritesRepo;
import timber.log.Timber;

public class WLApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
        ServiceLocator.wlDb = Room.databaseBuilder(getApplicationContext(), WLDatabase.class, getApplicationContext().getPackageName()).build();
        ServiceLocator.favoritesRepo = new FavoritesRepo(new HashSet<>(WLPreferences.loadStringSetPref(getApplicationContext(), Constants.PREF_IS_FAVORITE_KEY)));
    }
}

File: app/src/main/java/life/wanderinglocal/WLApplication.kt
package life.wanderinglocal

import android.app.Application
import androidx.room.Room
import life.wanderinglocal.repo.FavoritesRepo
import timber.log.Timber
import timber.log.Timber.DebugTree
import java.util.*

class WLApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }
        ServiceLocator.wlDb = Room.databaseBuilder(applicationContext, WLDatabase::class.java, applicationContext.packageName).build()
        ServiceLocator.favoritesRepo = FavoritesRepo(HashSet(WLPreferences.loadStringSetPref(applicationContext, Constants.PREF_IS_FAVORITE_KEY)))
    }
}